applications/StructuralMechanicsApplication/custom_elements/truss_elements/linear_truss_element_2D.h

```diff
@@ -0,0 +1,476 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:  Alejandro Cornejo
+//
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/element.h"
+#include "custom_utilities/structural_mechanics_element_utilities.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+using SizeType = std::size_t;
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class LinearTrussElement2D
+ * @ingroup StructuralMechanicsApplication
+ * @brief This is the Linear TRUSS element of 2 and 3 nodes.
+ * @author Alejandro Cornejo
+ */
+template <SizeType TNNodes>
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) LinearTrussElement2D
```

Comment:
whats the difference to the existing truss elements?

Reply:
These are 2D elements (thecurrent ones are just 3D). Besides, we use numerical integration so it is general in case a custom constitutive law is used.